app/domain/oidc/services/class.oidc.php

```diff
@@ -0,0 +1,398 @@
+<?php
```

Comment:
@TareqK Can you review this file in particular. I don't have enough knowledge with OIDC. To me it all looks good.

Reply:
It seems fine to me, should work on most providers. However, we need to add PSK verification since a lot of providers use rs256 PSK